src/python_testing/TC_EEVSE_Utils.py

```diff
@@ -0,0 +1,209 @@
+                                                      events=[(endpoint, self._expected_cluster, True)], reportInterval=(1, 5),
+                                                      fabricFiltered=False, keepSubscriptions=True, autoResubscribe=False)
+        self._subscription.SetEventUpdateCallback(self.__call__)
+
+    def __call__(self, res: EventReadResult, transaction: SubscriptionTransaction):
+        if res.Status == Status.Success and res.Header.ClusterId == self._expected_cluster.id:
+            logging.info(
+                f'Got subscription report for event on cluster {self._expected_cluster}: {res.Data}')
+            self._q.put(res)
+
+    def WaitForEventReport(self, expected_event: ClusterObjects.ClusterEvent):
+        try:
+            res = self._q.get(block=True, timeout=10)
+        except queue.Empty:
+            asserts.fail("Failed to receive a report for the event {}".format(expected_event))
+
+        asserts.assert_equal(res.Header.ClusterId, expected_event.cluster_id, "Expected cluster ID not found in event report")
+        asserts.assert_equal(res.Header.EventId, expected_event.event_id, "Expected event ID not found in event report")
+        return res.Data
+
+
+class EEVSEBaseTestHelper:
+
+    async def read_evse_attribute_expect_success(self, endpoint: int = None, attribute: str = ""):
+        full_attr = getattr(Clusters.EnergyEvse.Attributes, attribute)
+        cluster = Clusters.Objects.EnergyEvse
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=full_attr)
+
+    async def check_evse_attribute(self, attribute, expected_value, endpoint: int = None):
+        value = await self.read_evse_attribute_expect_success(endpoint=endpoint, attribute=attribute)
+        asserts.assert_equal(value, expected_value,
+                             f"Unexpected '{attribute}' value - expected {expected_value}, was {value}")
+
+    async def get_supported_energy_evse_attributes(self, endpoint: int = None):
+        return await self.read_evse_attribute_expect_success(endpoint, "AttributeList")
+
+    async def write_user_max_charge(self, endpoint: int = None, user_max_charge: int = 0):
+        if endpoint is None:
+            endpoint = self.matter_test_config.endpoint
+        result = await self.default_controller.WriteAttribute(self.dut_node_id,
+                                                              [(endpoint,
+                                                               Clusters.EnergyEvse.Attributes.UserMaximumChargeCurrent(user_max_charge))])
+        asserts.assert_equal(result[0].Status, Status.Success, "UserMaximumChargeCurrent write failed")
+
+    async def send_enable_charge_command(self, endpoint: int = None, charge_until: int = None, timedRequestTimeoutMs: int = 3000,
+                                         min_charge: int = None, max_charge: int = None, expected_status: Status = Status.Success):
```

Comment:
min charge and max charge aren't nullable, so setting the default to none here may cause weird errors. Is there a good default? 0?

Reply:
I think I left it as None so it would assert if these values weren't supplied, rather than sending 0 which may be treated as valid values and then potentially obscure an implementation error in the DUT if the test isn't good enough at checking the charging current used properly.

I'll refactor to ensure that they are always supplied by adding a test in the function to check it isn't None (rather than having to reorder the args which python insists on).

Reply:
Added follow up PR https://github.com/project-chip/connectedhomeip/issues/31460